cmd/awscollector/main_windows.go

```diff
@@ -44,6 +44,8 @@ func run(params otelcol.CollectorSettings) error {
 
 func runService(params otelcol.CollectorSettings) error {
 	// do not need to supply service name when startup is invoked through Service Control Manager directly
+	flagSet := config.Flags(featuregate.GlobalRegistry())
```

Comment:
Can this be hoisted even further into `main()` so that `params.ConfigProvider` has the correct value from the start?

Reply:
That's on the TODO list but I wasn't sure how safe it would be to do that. I would prefer to do this incrementally while we also look at further improvements to the collector startup sequence.